src/main/java/jcifs/dcerpc/msrpc/netdfs.java

```diff
@@ -0,0 +1,571 @@
+package jcifs.dcerpc.msrpc;
+
+
+import jcifs.dcerpc.DcerpcMessage;
+import jcifs.dcerpc.ndr.NdrBuffer;
+import jcifs.dcerpc.ndr.NdrException;
+import jcifs.dcerpc.ndr.NdrLong;
+import jcifs.dcerpc.ndr.NdrObject;
+
+
+@SuppressWarnings ( "all" )
+public class netdfs {
+
+    public static String getSyntax () {
+        return "4fc742e0-4a10-11cf-8273-00aa004ae673:3.0";
+    }
+
+    public static final int DFS_VOLUME_FLAVOR_STANDALONE = 0x100;
+    public static final int DFS_VOLUME_FLAVOR_AD_BLOB = 0x200;
+    public static final int DFS_STORAGE_STATE_OFFLINE = 0x0001;
+    public static final int DFS_STORAGE_STATE_ONLINE = 0x0002;
+    public static final int DFS_STORAGE_STATE_ACTIVE = 0x0004;
+
+    public static class DfsInfo1 extends NdrObject {
+
+        public String entry_path;
+
+
+        @Override
+        public void encode ( NdrBuffer _dst ) throws NdrException {
+            _dst.align(4);
+            _dst.enc_ndr_referent(this.entry_path, 1);
+
+            if ( this.entry_path != null ) {
+                _dst = _dst.deferred;
+                _dst.enc_ndr_string(this.entry_path);
+
+            }
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 31:38, 303:310.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.